enginetest/queries/priv_auth_queries.go

```diff
@@ -952,6 +952,105 @@ var UserPrivTests = []UserPrivilegeTest{
 			},
 		},
 	},
+	{
+		Name: "show user with no grants",
+		SetUpScript: []string{
+			"CREATE USER tester@localhost;",
+		},
+		Assertions: []UserPrivilegeTestAssertion{
+			{
+				User:  "root",
+				Host:  "localhost",
+				Query: "SHOW GRANTS FOR tester@localhost;",
```

Comment:
Any testing with `'%'` for the host that we should do? For example... I know if host is not specified, it defaults to `'%'`.

Reply:
I don't think we default to '%' if host isn't specified.

Reply:
[MySQL docs for show grants](https://dev.mysql.com/doc/refman/8.0/en/show-grants.html) say we should default to '%' if a hostname is omitted, so if we aren't doing that yet, it's worth matching the MySQL behavior.

Reply:
I see, right now we just use empty string if a host isn't provided.
Like `create user new_user` makes a `new_user@''`. I can add this behavior in a separate PR?

Reply:
Noice! Thanks for fixing that while you're in there so we can get a little closer to full MySQL compatibility. 😁 Tackling that in a separate PR sounds totally reasonable to me.